src/osd/ClassHandler.h

```diff
@@ -110,10 +109,13 @@ class ClassHandler
       const std::string& list);
 
 public:
+  Mutex mutex;
```

Comment:
@tchaikov I don't think that can be done easily. The issue is that call to `cls_init` methods immediately after `add_embedded_class` call `register_class` and that method expects the mutex to be already locked.